include/hipSYCL/std/stdpar/detail/offload.hpp

```diff
@@ -424,7 +424,7 @@ bool should_offload(AlgorithmType type, Size n, const Args &...args) {
       state.get_configuration().get_min_ops_per_offload_decision();
 
   int num_predicted_ops = 0;
-  uint64_t op_hash = get_operation_hash(type, n, args...);
+  uint64_t op_hash = get_operation_hash(type, args...);
```

Comment:
Yes, the idea was indeed to allow for the hash to include the problem size. We are not currently doing this because in the current implementation, we want to interpolate between problem sizes of one operation. Because of this, it is useful to store the operation data based on a hash without problem size, and then within the operation data store one entry per problem size.
However, I'm not sure if this is going to be the final way of doing things, so it might be a good idea to keep the problem size in the interface.

Reply:
I've added it back